internal/sorter/sorter.go

```diff
@@ -296,33 +297,44 @@ func (s routeSorter) Less(i, j int) bool {
 	switch a := s[i].PathMatchCondition.(type) {
 	case *dag.PrefixMatchCondition:
 		if b, ok := s[j].PathMatchCondition.(*dag.PrefixMatchCondition); ok {
-			cmp := strings.Compare(a.Prefix, b.Prefix)
```

Comment:
I think despite how any comments/docs may explain this, the important property that is being expressed here is that if there is a route with a prefix match of `/foo` and a route with a prefix match of `/foo/bar`, then the latter will sort first/take precedence when matching. In other words, more specific prefix matches take precedence over less specific ones. The current implementation actually implements that logic fine, since reverse lexicographic order will put more specific prefixes first.

Are there actually any scenarios where reverse lexicographic prefix match sorting does not give the desired result, but reverse length sorting does? As far as I can tell there shouldn't be, since for routing purposes we only care about relative ordering between pairs of prefixes where one wholly contains the other (e.g. `/foo/bar` vs. `/foo`).

Either way, I think the code change is fine/good since it's a little more clear, and will be more consistent with sorting for other match types, docs, comments, etc., but if the above is true then I would say the changelog should focus mainly on the behavior change for **regex** matching since that will definitely impact users. We can note that there will be some changes in prefix match ordering as well, but that they aren't expected to be impactful for routing behavior.

I can suggest some words for the changelog if we're on the same page about this, just want to get folks' input here first.

Reply:
What you are saying makes sense, I am just not sure if there is any case we might be missing

Reply:
fair point, I dont think I can think of anything that would affect this case